### ISSUE TITLE
iOS Quick Look Problem

### ISSUE THREAD

**Issue by sercandemirtas** [google/model-viewer#1521]
Hell all.

On ios quick look sometimes model is not show properly. It flashes. I dont know how to explain it. If you watch video I think you will understand. It start at sec. 2 and happens very quickly. What could be problem with that? 

https://drive.google.com/file/d/1ytuajLgqe5HFUggtzPk-jL_josW-SWW-/view

Thanks.

### OS
- [ ] Android
- [ x] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
I'm pretty sure that's the rendering getting out of sync with canvas resizing. Can you post a link to your site so I can try to repro? Also, what iOS device is this on?

Also, this is not a Quick Look problem, as your video is from before you enter AR mode.

**Comment by sercandemirtas** [comment]
I see you are right my bad it is not Quick Look problem. 

https://confident-lamport-09f645.netlify.app/m13.html

This problem is not always happens. I mean sometimes I wait 1 min and there is no problem but sometimes it happens after 3 second when we open page. 

My device is iphone 7 ios 13.6.1

**Comment by elalish** [comment]
Yeah, this is caused by the dynamic render sizing, and since that is performance-based, it'll repro differently depending on how hot your phone is, for instance. I'd like to call this a safari bug, but it's possible I can work around it. Meanwhile, if it really bothers you, you can disable it by setting the `minimumRenderScale` static property to 1, but this will come at the cost of frame rate drops.

**Comment by konradby** [comment]
@sercandemirtas @elalish this weird glitch also happens for Iphone 7 - Safari/Chrome/Firefox
eg. on this page:
https://caraya.github.io/model-viewer-demo/index.html

Setting `modelViewer.minimumRenderScale=1` didn't help at all - actually nothing really happened after changing this property, didn't notice any frame rate drop.

**Comment by elalish** [comment]
@konradby Thanks for the demo site, that's very nice! The minimumRenderScale is a static property, so you have to set it a little differently than attributes: `customElements.get("model-viewer").minimumRenderScale = 1;`. Does that make a difference?

**Comment by elalish** [comment]
@AdrianAtGoogle I think this is the bug you were noticing on iPad?

**Comment by AdrianAtGoogle** [comment]
- The issue in this bug (where for the bad frame it looks like the scale size is presented with the previous frame’s sampling bounds so the model visually pops to a smaller size) is not something I’ve seen directly. It looks bad; does it happen on every interaction? The only thing that jumps out as different than the stock MV examples which don’t seem to exhibit this behavior is I don’t think the examples are manipulating the mouse cursor icon; could updating that be introducing a frame of latency somehow?
- I didn’t know MV had dynamic render scale, it completely explains what I’ve mentioned observing in the safari debugger (which has a hall of mirrors style effect with the lower res renders painted over the higher res renders).  That said, IIRC it looked like the canvas debugger of a partially visible canvas seemed like it was rendering the full frame, not just the visible portion (I.e. the HoM effect was the full frame image at different sizes for a view that had only become partially visible)
- The flashing I’ve observed on iPad didn’t fit this profile, the cases all had backgrounds which don’t seem to flash at smaller scales. It seems like the view is struggling with the canvas size changing as the view scrolls on screen, but I haven’t been able to dig into it.
